### PR TITLE
Add documentation for the "unable to create new native thread" problem

### PR DESCRIPTION
**Pull request by bostko**
Added a Problems section in faq.md describing how to increase ulimit.

@rdowner please advice whether this is a good place for `increasing ulimit'.

### CODE REVIEW

File: docs/website/documentation/faq.md [hunk 1]
```diff
@@ -20,3 +20,29 @@ Supplying the answers are a TODO.
...
+To increase the limits, run `sudo vi /etc/security/limits.conf` and add (if it is "brooklyn" user running Apache Brooklyn):
+
+    brooklyn           soft    nproc           16384
+    brooklyn           hard    nproc           16384
+    brooklyn           soft    nofile          16384
+    brooklyn           hard    nofile          16384
+
+Then apply these changes by running `sudo sysctl -p`.
+
+Finally quit the ssh session, and log back in.
```
Comment (neykov) [inline docs/website/documentation/faq.md]: This looks like a distro specific config? Point out which distro it applies to.

File: docs/website/documentation/faq.md [hunk 2]
```diff
@@ -20,3 +20,29 @@ Supplying the answers are a TODO.
...
+## java.lang.OutOfMemoryError: unable to create new native thread
+
+You could encounter this error when running with many entities.
+
+Please **increase the ulimit** if you see such error:
+
+On the VM running Apache Brooklyn, we recommend ensuring nproc and nofile are reasonably high (e.g. higher than 1024, which is often the default).
+
+Run `ulimit -a` to see the current limits.
+
```
Comment (ygy) [inline docs/website/documentation/faq.md]: The correct setup for limits.conf is distribution specific. Maybe, it is better to point out only the recommended values without suggesting changing OS files and ask the user to consult distro's documentation.

File: docs/website/documentation/faq.md [hunk 3]
```diff
@@ -20,3 +20,29 @@ Supplying the answers are a TODO.
...
+To increase the limits, run `sudo vi /etc/security/limits.conf` and add (if it is "brooklyn" user running Apache Brooklyn):
+
+    brooklyn           soft    nproc           16384
+    brooklyn           hard    nproc           16384
+    brooklyn           soft    nofile          16384
+    brooklyn           hard    nofile          16384
+
+Then apply these changes by running `sudo sysctl -p`.
```
Comment (ygy) [inline docs/website/documentation/faq.md]: limits.conf is only read during login shells.

`sysctl -p` only enforces re-read of /etc/sysctl.conf

**Comment by ahgittin** [comment]
@bostko you have good comments from @ygy and @neykov.  it is useful to give os-specific commands but you should say what distro it is for.  i'd like to update and close this.

**Comment by aledsage** [comment]
Thanks @bostko - merging.

**Pull request merged**